python/tank/deploy/git_descriptor.py

```diff
@@ -259,8 +188,95 @@ def _find_latest_by_pattern(self, pattern):
             new_loc_dict,
             self._type
         )
-        
 
+    def _find_latest_tag_by_pattern(self, version_numbers, pattern):
```

Comment:
You're not missing anything obvious, quite the opposite!
I changed the function to fix it so it could support forked versions and version patterns, before that they were working but not both at the same time.
The original function was quite long (https://github.com/shotgunsoftware/tk-core/blob/34135_fw_versions/python/tank/deploy/git_descriptor.py#L192) and I changed it more than initially planned.
With hindsight, it looks like it might be easier to just use a regexp and search directly in the list of tags, instead of this 2-step approach (building a tree of versions, then browsing it).